interface/globals.php

```diff
@@ -596,6 +597,23 @@
 $GLOBALS['customModDir'] = "custom_modules"; //non zend modules
 $GLOBALS['zendModDir'] = "zend_modules"; //zend modules
 
+try {
+    // load up the modules system and bootstrap them.
+    // This has to be fast, so any modules that tie into the bootstrap must be kept lightweight
+    // registering event listeners, etc.
+    // TODO: why do we have 3 different directories we need to pass in for the zend dir path. shouldn't zendModDir already have all the paths set up?
+    /** @var ModulesApplication */
+    $GLOBALS['modules_application'] = new ModulesApplication(
+        $GLOBALS["kernel"],
+        $GLOBALS['fileroot'],
+        $GLOBALS['baseModDir'],
+        $GLOBALS['zendModDir']
+    );
+} catch (\Exception $ex) {
+    error_log($ex->getMessage() . $ex->getTraceAsString());
```

Comment:
use log escape:
```
error_log(errorLogEscape($ex->getMessage() . $ex->getTraceAsString()));
```

Reply:
Good catch, I forget that we can have exceptions triggered by bad callees :)

Reply:
lol, I don't even think about these things anymore. If it's a variable, I just escape it. Even it the variable was created 1 line above from a known string, I would still escape it; utter laziness :)